understanding/sharding.md

```diff
@@ -2,29 +2,210 @@
+
+Here's an example of a function that uses `fetchClientValues()` to first get, then add the total number of guilds from *all shards* (i.e. your bot's total guild count):
+
+```javascript
+/*
+  	Example by ZiNc#2032
+  	The following code fetches total combined shards' server counts.
+  
+  	discord.js version 12.x
+  	client = new discordjs.Client()
+
+  	returns number
+*/
+
+const getServerCount = async () => {
+    // get guild collection size from all the shards
+    const req = await client.shard.fetchClientValues('guilds.cache.size');
+
+    // return the added value
+    return req.reduce((p, n) => p + n, 0);
+}
+```
+
+{% hint style="info" %}
+`fetchClientValues()` does not allow you to make use of javascript methods or client methods to get or process information before returning it. It only allows you to get information from client properties.
+{% endhint %}
+
+### BroacastEval
+
+[`broadcastEval`](https://discord.js.org/#/docs/main/v12/class/ShardClientUtil?scrollTo=broadcastEval) evaluates the input in the context of each shard's Client(s). This is what you should use when you want to execute a method or process data on a shard and return the result. It's useful for getting information that isn't available through client properties and must instead be retrieved through the use of methods.
+
+Example:
+```javascript
+/*
+    Example of result of broadcastEval() on a bot with 4 servers split across 2 shards.
+    Assume this is being executed on shard 0, the first shard.
+*/
+
+// If we just map our guilds' members.cache.size, it will return
+// only the mapped members.size of the shard this is being run on.
+console.log(client.guilds.cache.map((guild) => guild.members.cache.size));
+//		[
+//			30,
+//			25
+//		],
+
+// If we would like to map our guilds' members.cache.size from our
+// servers on all of our shards, we must make use of broadcastEval().
+// Remember, this runs in the context of the client, so we refer to the
+// Client using "this".
+const res = await client.shard.broadcastEval('this.guilds.cache.map((guild) => guild.members.cache.size)');
+console.log(res);
+// 	[
+//		[	// shard 0
+//			30,
+//			25
+//		],
+//		[	// shard 1
+//			55,
+//			10
+//		]
+// 	]
+
+````
+
+Say you want to get a guild from your client. In a non-sharded environment, you would simply use `client.guilds.cache.get('ID')` or something of that nature and then carry on with your code. In this case however, it is possible that the guild you're trying to get *is not present on the shard*. In order to get the guild for use, you would then need to fetch it from whatever shard it is present on using `broadcastEval()`.
```

Comment:
This is, again, redundant because it says the same thing as the code samples above.